Zend/zend_compile.c

```diff
@@ -4718,19 +4718,19 @@ static void zend_compile_rope_finalize(znode *result, uint32_t j, zend_op *init_
 
 static zend_result zend_compile_func_sprintf(znode *result, zend_ast_list *args) /* {{{ */
 {
-	znode format_string;
-
 	if (args->children < 1) {
 		return FAILURE;
 	}
 
-	zend_compile_expr(&format_string, args->child[0]);
-	if (format_string.op_type != IS_CONST || Z_TYPE(format_string.u.constant) != IS_STRING) {
-		zend_do_free(&format_string);
+	zend_eval_const_expr(&args->child[0]);
+	if (!args->child[0] || args->child[0]->kind != ZEND_AST_ZVAL) {
```

Comment:
I don't think you need the `NULL` check?

Reply:
Indeed. That is probably a left-over from my attempts of getting this to work. I've initially used `args->child[1]` by accident and obviously got crashes then …